lib/stdlib/src/lists.erl

```diff
@@ -727,6 +731,32 @@ delete(Item, [H|Rest]) ->
     [H|delete(Item, Rest)];
 delete(_, []) -> [].
 
+-doc """
+Returns a copy of `List1` where all elements matching `Elem` are removed, if
+there are any such elements.
+
+## Examples
+
+```erlang
+1> lists:deleteall(x, [a, x, b, x, c]).
+[a,b,c]
+2> lists:deleteall(x, [d, e, f]).
+[d,e,f]
+```
+""".
+-spec deleteall(Elem, List1) -> List2 when
+      Elem :: T,
+      List1 :: [T],
+      List2 :: [T],
+      T :: term().
+
+deleteall(Item, [Item|Rest]) ->
+    deleteall(Item, Rest);
+deleteall(Item, [H|Rest]) ->
+    [H|deleteall(Item, Rest)];
+deleteall(_, []) ->
+    [].
```

Comment:
Any reason for not using a one-liner?

```Erlang
deleteall(Elem, List) ->
    [E || E <- List, E =/= Elem].
```

Reply:
Ok, no problem :smile_cat: Is that a final verdict? Then I'll close this PR.

Reply:
On a side note: The reason why I am kinda keen on having dedicated functions for everything is that my IDE allows me to find and highlight function usages easily, while it would be rather hard to find certain list comprehensions like the one above that do what a `deleteall` function does. So in the end, I would _still_ want to put a `deleteall` function in utility modules in my projects. (Just my 2ct :wink:)

Reply:
> Any reason for not using a one-liner?
> 
> ```erlang
> deleteall(Elem, List) ->
>     [E || E <- List, E =/= Elem].
> ```

Sure, why not :smiley_cat: I was following the pattern of `delete`. I'll wait for some more opinions if this is actually wanted or not, and change it to a list comprehension if yes :wink: